andela-sjames/Django-ReactJS-Library-App
# Feature/user authentication

#### What Does This PR Do?

#### Description Of Task To Be Completed
- Server-side view should make use of the Google OAuth Library
- Server-side should return a response to the client with a JWT encoded
   payload as well a user object.
- Server-side view should ensure that the user is of Andela G suite

#### Any Background Context You Want To Provide?
Currently, there is no form of token authentication for the server-side and calls can be made to any endpoint without token verification. 
#### How can this be manually tested?
Using Postman, enter the url: `localhost:8000/api/v1/auth/register/` with post data
`{"idtoken": "random_id_token_from_google!",}` to the url. 
media type should be `x-www-form-urlencoded` 
#### Screenshot(s)
N/A

#### What are the relevant GitHub issues?
